src/target/riscv/riscv.c

```diff
@@ -3916,6 +4010,13 @@ int riscv_enumerate_triggers(struct target *target)
 					riscv_set_register(target, GDB_REGNO_TDATA1, 0);
 				break;
 		}
+
+		uint64_t tinfo;
+		result = riscv_get_register(target, &tinfo, GDB_REGNO_TINFO);
+		if (result == ERROR_OK)
+			r->trigger_tinfo[t] = tinfo;
+		else
+			r->trigger_tinfo[t] = 1 << type;
 	}
```

Comment:
Please, print the types supported in each trigger. This will be very useful when debugging issues.

For example:

```suggestion

		LOG_TARGET_DEBUG(target, "Trigger %u: supported types (mask) = 0x%08x", t, r->trigger_tinfo[t]);
	}
```